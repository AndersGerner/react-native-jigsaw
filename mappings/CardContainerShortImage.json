[{"name":"Short Card (Left)","tag":"CardContainerShortImage","description":"An elevated card with a left aligned image and a title and subtitle","category":"card","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Card_ContainerShortLeftImage.png","supports_list_render":true,"props":{"image":{"label":"Image","description":"Image","type":"remoteImage","value":null,"editable":true},"title":{"label":"Title","description":"Title to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"subtitle":{"label":"Subtitle","description":"Subtitle to display","type":"string","value":"San Diego","editable":true},"aspectRatio":{"label":"Aspect ratio","description":"Aspect ratio of the image","type":"aspectRatio","value":null,"editable":true,"required":false},"elevation":{"label":"Elevation","description":"Elevation of the card. A number 0-3.","type":"number","value":2,"min":0,"max":3,"step":1,"precision":0,"editable":true},"mode":{"type":"string","value":"left","editable":false}},"layout":{}},{"name":"Short Card (Right)","tag":"CardContainerShortImage","description":"An elevated card with a right aligned image and a title and subtitle","category":"card","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Card_ContainerShortRightImage.png","supports_list_render":true,"props":{"image":{"label":"Image","description":"Image","type":"remoteImage","value":null,"editable":true},"title":{"label":"Title","description":"Title to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"subtitle":{"label":"Subtitle","description":"Subtitle to display","type":"string","value":"San Diego","editable":true},"elevation":{"label":"Elevation","description":"Elevation of the card. A number 0-3.","type":"number","value":2,"min":0,"max":3,"step":1,"precision":0,"editable":true},"mode":{"type":"string","value":"right","editable":false}},"layout":{}}]