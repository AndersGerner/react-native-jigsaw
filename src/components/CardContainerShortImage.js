import React from "react";
import { Image, View, Text } from "react-native";
import Card from "./Card";
import Elevation from "./Elevation";
import { withTheme } from "../core/theming";
import { COMPONENT_TYPES, FORM_TYPES } from "../core/component-types";
import type { Theme } from "../types";
import Config from "./Config";

export type CardContainerShortImageProps = {
  image: string | { uri: string },
  title?: string,
  subtitle?: string,
  aspectRatio?: number,
  mode: "left" | "right",
  elevation: number,
  theme: Theme,
  style: any,
  onPress: () => void
};

class CardContainerShortImage extends React.PureComponent<CardContainerShortImageProps> {
  static defaultProps = {
    elevation: 2,
    aspectRatio: 1,
    mode: "left"
  };

  render() {
    const {
      image,
      title,
      subtitle,
      mode,
      aspectRatio,
      elevation,
      theme: { colors, borderRadius, typography, spacing },
      style,
      onPress
    } = this.props;

    return (
      <Card style={style} onPress={onPress}>
        <Elevation
          style={{
            elevation,
            borderRadius: borderRadius.global
          }}
        >
          <View
            style={{
              overflow: "hidden",
              flexDirection: "row",
              justifyContent: mode === "right" ? "space-between" : "flex-start",
              borderRadius: borderRadius.global
            }}
          >
            {mode === "left" && (
              <Image
                style={{ aspectRatio }}
                source={typeof image === "string" ? { uri: image } : image}
                resizeMode="cover"
              />
            )}
            <View
              style={{
                padding: spacing.large,
                backgroundColor: colors.surface,
                flex: 1
              }}
            >
              <Text
                numberOfLines={1}
                style={[typography.headline5, { color: colors.strong }]}
              >
                {title}
              </Text>
              {subtitle ? (
                <Text
                  numberOfLines={1}
                  style={[
                    typography.body2,
                    { color: colors.medium, marginTop: spacing.text }
                  ]}
                >
                  {subtitle}
                </Text>
              ) : null}
            </View>
            {mode === "right" && (
              <Image
                style={{ aspectRatio }}
                source={typeof image === "string" ? { uri: image } : image}
                resizeMode="cover"
              />
            )}
          </View>
        </Elevation>
      </Card>
    );
  }
}

export default withTheme(CardContainerShortImage);

export const SEED_DATA = [
  {
    name: "CardContainerShortLeftImage",
    tag: "CardContainerShortImage",
    description:
      "An elevated card with a left aligned image and a title and subtitle",
    category: COMPONENT_TYPES.card,
    preview_image_url:
      "https://res.cloudinary.com/altos/image/upload/v1541096710/draftbit/library/jigsaw-1.0/reps/Card_ContainerShortLeftImage.png",
    supports_list_render: true,
    props: {
      image: {
        label: "Image",
        description: "Image",
        type: FORM_TYPES.remoteImage,
        value: null,
        editable: true
      },
      title: {
        label: "Title",
        description: "Title to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      subtitle: {
        label: "Subtitle",
        description: "Subtitle to display",
        type: FORM_TYPES.string,
        value: "San Diego",
        editable: true
      },
      aspectRatio: {
        label: "Aspect ratio",
        description: "Aspect ratio of the image",
        type: FORM_TYPES.aspectRatio,
        value: null,
        editable: true,
        required: false
      },
      elevation: {
        label: "Elevation",
        description: "Elevation of the card. A number 0-3.",
        type: FORM_TYPES.number,
        value: 2,
        min: 0,
        max: 3,
        step: 1,
        precision: 0,
        editable: true
      },
      mode: {
        type: FORM_TYPES.string,
        value: "left",
        editable: false
      }
    },
    layout: {
      width: 345,
      height: 84
    }
  },
  {
    name: "CardContainerShortRightImage",
    tag: "CardContainerShortRImage",
    description:
      "An elevated card with a right aligned image and a title and subtitle",
    category: COMPONENT_TYPES.card,
    preview_image_url:
      "https://res.cloudinary.com/altos/image/upload/v1541096716/draftbit/library/jigsaw-1.0/reps/Card_ContainerShortRightImage.png",
    supports_list_render: true,
    props: {
      image: {
        label: "Image",
        description: "Image",
        type: FORM_TYPES.remoteImage,
        value: null,
        editable: true
      },
      title: {
        label: "Title",
        description: "Title to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      subtitle: {
        label: "Subtitle",
        description: "Subtitle to display",
        type: FORM_TYPES.string,
        value: "San Diego",
        editable: true
      },
      elevation: {
        label: "Elevation",
        description: "Elevation of the card. A number 0-3.",
        type: FORM_TYPES.number,
        value: 2,
        min: 0,
        max: 3,
        step: 1,
        precision: 0,
        editable: true
      },
      mode: {
        type: FORM_TYPES.string,
        value: "right",
        editable: false
      }
    },
    layout: {
      width: 375,
      height: 84
    }
  }
];
