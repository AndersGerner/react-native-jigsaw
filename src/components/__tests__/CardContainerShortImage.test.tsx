import * as React from "react";
import renderer from "react-test-renderer";
import CardContainerShortImage from "../CardContainerShortImage";

it("renders card container short image", () => {
  const tree = renderer
    .create(<CardContainerShortImage onPress={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
