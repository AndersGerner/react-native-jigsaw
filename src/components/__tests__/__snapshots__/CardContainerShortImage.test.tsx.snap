// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card container short image 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 718,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderColor": "rgba(18, 20, 44, 0)",
          "borderRadius": 6,
          "borderWidth": 0,
          "elevation": 2,
          "shadowColor": "rgba(18, 20, 44, 1)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.0799999982,
          "shadowRadius": 4,
          "zIndex": 100,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 6,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "aspectRatio": 1,
              },
            ]
          }
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "https://res.cloudinary.com/altos/image/upload/c_scale,q_auto,dpr_auto,w_100,h_100/Avatar.png",
              }
            }
            style={
              Array [
                Object {
                  "aspectRatio": 1,
                },
                Object {
                  "aspectRatio": undefined,
                },
                Object {
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "flex": 1,
              "padding": 16,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "fontFamily": "System",
                  "fontSize": 20,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 26,
                },
                Object {
                  "color": "rgba(18, 20, 44, 1)",
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
